test/runtests.jl

```diff
@@ -320,6 +323,55 @@ function test_tracer_diagnostics(model)
 
     return nothing
 end
+
+function test_density_equation_terms_errors(model)
+
+    @test_throws ArgumentError GravitationalPotentialEnergy(model)
+    @test_throws ArgumentError GravitationalPotentialEnergy(model, 0)
+
+    return nothing
+end
+
+function test_density_equation_terms(model)
+
+    Eₚ = GravitationalPotentialEnergy(model)
+    Eₚ_field = Field(Eₚ)
+    @test Eₚ isa AbstractOperation
+    @test Eₚ_field isa Field
+    compute!(Eₚ_field)
+
+    ρ = Field(seawater_density(model))
+    compute!(ρ)
+    Z = Field(model_geopotential_height(model))
+    compute!(Z)
+    g = model.buoyancy.model.gravitational_acceleration
+
+    true_value =  g .* ρ.data .* Z.data
+
+    @test isequal(Eₚ_field.data, true_value)
+
+    return nothing
+end
+function test_density_equation_terms(model, reference_geopotential_height)
+
+    Eₚ = GravitationalPotentialEnergy(model, reference_geopotential_height)
+    Eₚ_field = Field(Eₚ)
+    @test Eₚ isa AbstractOperation
+    @test Eₚ_field isa Field
+    compute!(Eₚ_field)
+
+    ρ = Field(seawater_density(model, geopotential_height = reference_geopotential_height))
+    compute!(ρ)
+    Z = Field(model_geopotential_height(model))
+    compute!(Z)
+    g = model.buoyancy.model.gravitational_acceleration
+
+    true_value =  @. g * ρ.data * Z.data
+
+    @test isequal(Eₚ_field.data, true_value)
```

Comment:
Can we merge these into one function?

Reply:
Yep, depends on how [merging the functions is done](https://github.com/tomchor/Oceanostics.jl/pull/164#discussion_r1536057285) but will update this to one function after a decision there.

Reply:
Actually this can be done with `kwarg` and conditional so have done so.